Samples/CayenneDecoder/Classes/LoraDecoders.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Text;
+using Newtonsoft.Json;
+
+namespace CayenneDecoderModule.Classes
+{
+    internal static class LoraDecoders
+    {
+        private static string CayenneDecoder(byte[] payload, uint fport)
+        {
+            CayenneDecoder cayenneDecoder = new CayenneDecoder(payload);
+            CayenneDevice cayenneDevice = new CayenneDevice();
+
+            if (cayenneDecoder.TryGetAnalogInput(out AnalogInput analogInput))
```

Comment:
Is the `next byte` to read telling us the type of the payload? If that is the case can't we make it faster and create the correct type instead of trying all types?

Reply:
if there was only 1 devices to decode, then yes. But most payload contains multiple devices. And they is no specific order to put them one after the other. And as the size is different every time, then you can't use this method. Yes, if there were thousands of devices, there maybe few other ways to optimize and cache some decoded information. But in this case, there are max like 4 or 5. So it goes actually largely fast enough.

Reply:
Just trying to see if I understand the logic. Would the snippet below work to decode multiple values?

```c#
byte[] input = GetInputFromSomewhere();

var index = 0;
while (index < input.Length)
{
   switch (input[index])
   {
      case (int)CayenneTypes.DigitalInput:
          // create Digital input starting from input[index]
          index += (int)CayenneTypeSize.DigitalInput;
          break;

     // add here all other types && default handling to stop
   }   
}

```

Reply:
The problem with the way the packet is formed is that the it is a sum of chuncks which have various sizes. And the first element of the table is unfortunately not the type of input. It is the second element. This second element gives you the type of sensor which gives you the full size (including the first element and the type of sensor) of the chunck. 
A code like you propose can work but is much longer than the one with the matching enums and extracting the real data. 
Keep in mind that once you have the real data, they all need to be transformed differently.
The way I can kind of improve the performance if we really want Something with more performance if to create a list of devices and parse only once the full buffer.

Reply:
Thanks for the explanation. I am not worried about performance, I am thinking about a cleaner way to accomplish it without the "ifs".